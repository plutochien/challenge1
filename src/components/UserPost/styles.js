import styled from 'styled-components';

export const UserPostContainer = styled.div`
  margin-top: 150px;
`;

export const Text = styled.span`
  display: flex;
  padding: 10px;
`;
