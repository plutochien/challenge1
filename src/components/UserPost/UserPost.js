import React, { Fragment, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { fetchPosts } from '../../actions/postAction';
import { connect } from 'react-redux';
import { UserPostContainer, Text } from './styles';
import Spinner from '../Spinner/Spinner';
import Post from '../Post/Post';
import { Button } from '../Post/styles';
import { SHOW_ALL } from '../../util/constatns';
import ArrowIcon from '../../assets/icons/expand_arrow.svg';

const UserPost = ({
  userId,
  posts,
  fetchPosts,
  isPostLoading,
  currentPostId,
}) => {
  const [showAll, setShowAll] = useState(false);
  useEffect(() => {
    fetchPosts(userId);
  }, [userId, fetchPosts]);

  const handleClick = event => {
    event.preventDefault();
    setShowAll(true);
  };

  const renderPostList = () => {
    return (
      <Fragment>
        {posts.map((post, index) => {
          if (!showAll && index > 2) {
            return null;
          }
          return (
            <Post key={post.id} post={post} currentPostId={currentPostId} />
          );
        })}
        {!showAll && (
          <Button onClick={handleClick}>
            <img src={ArrowIcon} alt="show all" /> <Text>{SHOW_ALL}</Text>
          </Button>
        )}
      </Fragment>
    );
  };
  return (
    <UserPostContainer>
      <Fragment>
        {isPostLoading && <Spinner />}
        {renderPostList()}
      </Fragment>
    </UserPostContainer>
  );
};

UserPost.propTypes = {
  userId: PropTypes.string.isRequired,
  posts: PropTypes.array.isRequired,
  fetchPosts: PropTypes.func.isRequired,
  isPostLoading: PropTypes.bool,
  currentPostId: PropTypes.number,
};

const mapStateToProps = (state, ownProps) => ({
  isPostLoading: state.postsReducer.isPostLoading,
  currentPostId: state.commentsReducer.currentPostId,
  posts: state.postsReducer.posts,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      fetchPosts,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(UserPost);
