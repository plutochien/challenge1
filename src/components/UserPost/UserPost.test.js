import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import UserPost from "./UserPost";
const mockStore = configureStore([]);

describe('UserPost Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      usersReducer: {
        users: [
          {
            id: 1,
            name: 'Leanne Graham',
            username: 'Bret',
            email: 'Sincere@april.biz',
            phone: '1-770-736-8031 x56442',
            website: 'hildegard.org',
            company: {
              name: 'Romaguera-Crona',
              catchPhrase: 'Multi-layered client-server neural-net',
              bs: 'harness real-time e-markets',
            },
          },
        ],
        isUserLoading: false,
      },
      postsReducer: {
        posts: [
          {
            userId: 4,
            id: 31,
            title: 'ullam ut quidem id aut vel consequuntur',
            body: 'debitis eius sed qadipisci repudiandae',
          },
        ],
        isPostLoading: false,
      },
      commentsReducer: {
        comments: [
          {
            postId: 32,
            id: 156,
            name: 'pariatur omnis in',
            email: 'Telly_Lynch@karl.co.uk',
            body: 'dolorum voluptas laborinus dolores magni',
          },
        ],
        isCommentsLoading: false,
        currentPostId: 31,
      },
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <UserPost userId={1} />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
