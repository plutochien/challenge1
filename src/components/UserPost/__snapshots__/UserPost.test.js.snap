// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPost Component should render with given state from Redux store 1`] = `
<div
  className="sc-bdfBwQ gcIgTB"
>
  <div
    className="sc-dlfnbm kafrza"
  >
    <div
      className="sc-hKgILt dkOHOD"
    >
      ullam ut quidem id aut vel consequuntur
    </div>
    <div
      className="sc-eCssSg cGrduj"
    >
      debitis eius sed qadipisci repudiandae
    </div>
    <button
      className="sc-jSgupP dZKGaM"
      onClick={[Function]}
    >
      <img
        alt="expand"
        src="expand_arrow.svg"
      />
    </button>
  </div>
  <button
    className="sc-jSgupP dZKGaM"
    onClick={[Function]}
  >
    <img
      alt="show all"
      src="expand_arrow.svg"
    />
     
    <span
      className="sc-gsTCUz jqfHj"
    >
      show all
    </span>
  </button>
</div>
`;
